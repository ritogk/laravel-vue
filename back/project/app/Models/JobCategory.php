<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use DateTimeInterface;

/**
 * App\Models\JobCategory
 *
 * @property int $id
 * @property string $name 名称
 * @property string $content 内容
 * @property string $image 画像パス
 * @property int $sort_no 並び順
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 * @property \Illuminate\Support\Carbon|null $deleted_at
 * @property-read \Illuminate\Database\Eloquent\Collection|\App\Models\Job[] $job
 * @property-read int|null $job_count
 * @method static \Database\Factories\JobCategoryFactory factory(...$parameters)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory newModelQuery()
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory newQuery()
 * @method static \Illuminate\Database\Query\Builder|JobCategory onlyTrashed()
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory query()
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereContent($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereCreatedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereDeletedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereId($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereImage($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereName($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereSortNo($value)
 * @method static \Illuminate\Database\Eloquent\Builder|JobCategory whereUpdatedAt($value)
 * @method static \Illuminate\Database\Query\Builder|JobCategory withTrashed()
 * @method static \Illuminate\Database\Query\Builder|JobCategory withoutTrashed()
 * @mixin \Eloquent
 */
class JobCategory extends Model
{
    use SoftDeletes, HasFactory;

    protected $guarded = ['id', 'updated_at', 'created_at', 'deleted_at']; // ブラックリスト

    // リレーション
    public function job()
    {
        return $this->hasMany('App\Models\Job');
    }

    /**
    * 日時を日本時間に変換する
    * laravel7からtoArray, toJson内の日時がUTC固定になったので
    *
    * @param  \DateTimeInterface  $date
    * @return string
    */
   protected function serializeDate(DateTimeInterface $date)
   {
       return $date->format('Y-m-d H:i:s');
   }
}
