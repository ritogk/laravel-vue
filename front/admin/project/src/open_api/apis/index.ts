/* tslint:disable */
/* eslint-disable */
export * from './AuthAdminApi';
export * from './AuthFrontApi';
export * from './EntryApi';
export * from './FileApi';
export * from './JobApi';
export * from './JobCategorieApi';
export * from './UserApi';
