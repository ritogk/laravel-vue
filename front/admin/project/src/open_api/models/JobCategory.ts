/* tslint:disable */
/* eslint-disable */
/**
 * OpenAPI Tutorial
 * OpenAPI Tutorial by halhorn
 *
 * The version of the OpenAPI document: 0.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 職種
 * @export
 * @interface JobCategory
 */
export interface JobCategory {
    /**
     * id
     * @type {number}
     * @memberof JobCategory
     */
    id: number;
    /**
     * 名称
     * @type {string}
     * @memberof JobCategory
     */
    name: string;
    /**
     * 内容
     * @type {string}
     * @memberof JobCategory
     */
    content: string;
    /**
     * 画像の内部識別子
     * @type {string}
     * @memberof JobCategory
     */
    image: string;
    /**
     * 画像URL
     * @type {string}
     * @memberof JobCategory
     */
    imageUrl: string;
    /**
     * 並び順
     * @type {number}
     * @memberof JobCategory
     */
    sortNo: number;
    /**
     * 作成日時
     * @type {Date}
     * @memberof JobCategory
     */
    createdAt?: Date;
    /**
     * 更新日時
     * @type {Date}
     * @memberof JobCategory
     */
    updatedAt?: Date;
}

export function JobCategoryFromJSON(json: any): JobCategory {
    return JobCategoryFromJSONTyped(json, false);
}

export function JobCategoryFromJSONTyped(json: any, ignoreDiscriminator: boolean): JobCategory {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': json['id'],
        'name': json['name'],
        'content': json['content'],
        'image': json['image'],
        'imageUrl': json['imageUrl'],
        'sortNo': json['sortNo'],
        'createdAt': !exists(json, 'createdAt') ? undefined : (new Date(json['createdAt'])),
        'updatedAt': !exists(json, 'updatedAt') ? undefined : (new Date(json['updatedAt'])),
    };
}

export function JobCategoryToJSON(value?: JobCategory | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'id': value.id,
        'name': value.name,
        'content': value.content,
        'image': value.image,
        'imageUrl': value.imageUrl,
        'sortNo': value.sortNo,
        'createdAt': value.createdAt === undefined ? undefined : (value.createdAt.toISOString().substr(0,10)),
        'updatedAt': value.updatedAt === undefined ? undefined : (value.updatedAt.toISOString().substr(0,10)),
    };
}

