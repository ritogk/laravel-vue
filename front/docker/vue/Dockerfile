FROM node:14.19.0-buster

WORKDIR /usr/src/app

RUN apt-get update \
&& apt-get install -y \
&& npm install \