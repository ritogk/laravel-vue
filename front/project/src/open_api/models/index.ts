/* tslint:disable */
/* eslint-disable */
export * from './AccessToken';
export * from './Admin';
export * from './Entry';
export * from './Job';
export * from './JobCategory';
export * from './ModelFile';
export * from './QueryEntryList';
export * from './QueryJobCategoryList';
export * from './QueryJobList';
export * from './QueryUserList';
export * from './RequestEntry';
export * from './RequestFile';
export * from './RequestJob';
export * from './RequestJobCategory';
export * from './RequestLogin';
export * from './RequestUser';
export * from './Sex';
export * from './User';
