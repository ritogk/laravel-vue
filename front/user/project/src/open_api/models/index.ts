/* tslint:disable */
/* eslint-disable */
export * from './AccessToken';
export * from './Admin';
export * from './Entry';
export * from './FilePath';
export * from './Job';
export * from './JobCategory';
export * from './QueryEntryList';
export * from './QueryJobCategoryList';
export * from './QueryJobList';
export * from './QueryUserList';
export * from './RequestEntry';
export * from './RequestJob';
export * from './RequestJobCategory';
export * from './RequestLogin';
export * from './RequestUser';
export * from './ResponseJobEntry';
export * from './Sex';
export * from './User';
