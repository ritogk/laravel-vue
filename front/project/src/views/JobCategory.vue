<style scoped>
img {
}
</style>

<template>
  <div class="p-4">
    <div class="card border-primary mb-3">
      <div class="card-header bg-primary text-white">職種</div>
      <div class="card-body">
        <h6 class="card-subtitle mb-2 text-muted">
          ご希望の職種を選択して下さい。
        </h6>
        <div class="d-flex flex-wrap">
          <div
            v-for="(category, index) in categories"
            :key="`category-${index}`"
            :category="category"
            v-show="!loading"
          >
            <div class="card text-white">
              <img
                :src="`http://localhost${category.imageUrl}`"
                class="card-img category-card"
                style="
                  max-width: 20rem;
                  height: 200px;
                  filter: brightness(0.65);
                "
              />
              <div class="card-img-overlay">
                <h4 class="card-title" v-text="category.name"></h4>
                <p class="card-text">
                  <span v-text="category.content"></span>
                  <a
                    href="#"
                    @click.prevent="pageTransition(category)"
                    class="stretched-link"
                  ></a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import {
  computed,
  defineComponent,
  reactive,
  watchEffect,
  onMounted,
  ref,
} from 'vue';
import { JobCategorieApi } from '@/open_api/apis/JobCategorieApi';
import { JobCategory } from '@/open_api';

export default defineComponent({
  setup() {
    const categories = ref<JobCategory[]>([]);
    const loading = ref(true);

    const jobCategoryApi = new JobCategorieApi();

    const load = async () => {
      const response = await jobCategoryApi.jobCategoriesGet({});
      categories.value = response;
      loading.value = false;
    };
    load();

    return { categories, loading };
  },
});
</script>
