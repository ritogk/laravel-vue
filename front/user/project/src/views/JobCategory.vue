<template>
  <div class="card border-primary mb-3">
    <div class="card-header bg-primary text-white">職種</div>
    <div class="card-body">
      <h6 class="card-subtitle mb-2 text-muted">
        ご希望の職種を選択して下さい。(ci_test)
      </h6>
      <div class="d-flex flex-wrap mb-3">
        <VueElementLoading :active="loading" spinner="ring" size="50" />
        <div
          v-for="(category, index) in categories"
          :key="`category-${index}`"
          :category="category"
          v-show="!loading"
          class="m-2"
        >
          <div @click.prevent="clickCategory(category.id)">
            <TextOnImage
              :imageUrl="category.imageUrl"
              :title="category.name"
              :text="category.content"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue';
import { useRouter } from 'vue-router';
import TextOnImage from '@/components/TextOnImage.vue';
import VueElementLoading from 'vue-element-loading';
import { useJobCategory } from '@/composables/useJobCategory';

export default defineComponent({
  components: {
    VueElementLoading,
    TextOnImage,
  },
  setup() {
    const router = useRouter();
    const jobCategory = useJobCategory();
    const loading = ref(true);
    const categories = jobCategory.jobCategoryRefs.items;

    // 読み込み
    const load = async () => {
      await jobCategory.getJobCategory();
      loading.value = false;
    };
    load();

    // 「職種」押下時の処理
    const clickCategory = (categoryId: string) => {
      router.push({
        name: 'JobEntry',
        query: { jobCategoryId: categoryId },
      });
    };

    return { categories, loading, clickCategory };
  },
});
</script>
