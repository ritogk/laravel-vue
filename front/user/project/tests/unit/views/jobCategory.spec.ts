import { nextTick } from 'vue';
import { flushPromises, mount, VueWrapper } from '@vue/test-utils';
import * as vueRouter from 'vue-router';
import { JobCategorieApi } from '@/open_api';
import { jobCategoryModel1, jobCategoryModel2 } from '~/utils/data/jobCategory';
// テスト対象のコンポーネント
import JobCategory from '@/views/JobCategory.vue';

describe('職種選択画面で正しい内容が表示されているか', () => {
  let wrapper: VueWrapper<any>;

  /**
   * 外部からの振る舞いを設定
   */
  beforeEach(async () => {
    // モックを作らないとなぜかエラーがでる
    jest
      .spyOn(vueRouter, 'useRouter')
      .mockImplementation((): any => ({ push: () => {} }));

    // 職種一覧取得関数のモックを作成
    jest
      .spyOn(JobCategorieApi.prototype, 'jobCategoriesGet')
      .mockImplementation(() =>
        Promise.resolve([jobCategoryModel1, jobCategoryModel2])
      );

    // コンポーネントをレンダリング
    wrapper = mount(JobCategory);

    await flushPromises();
    await nextTick();
  });

  /**
   * 確認項目
   */
  it('スナップショットとの差分を確認する', async () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
