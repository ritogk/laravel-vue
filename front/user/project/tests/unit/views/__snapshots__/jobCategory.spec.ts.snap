// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`職種選択画面で正しい内容が表示されているか スナップショットとの差分を確認する 1`] = `
<div
  class="card border-primary mb-3"
>
  <div
    class="card-header bg-primary text-white"
  >
    職種
  </div>
  <div
    class="card-body"
  >
    <h6
      class="card-subtitle mb-2 text-muted"
    >
       ご希望の職種を選択して下さい。aaaa 
    </h6>
    <div
      class="d-flex flex-wrap mb-3"
    >
      <transition-stub
        data-v-2964abc9=""
      >
        <div
          class="velmld-overlay"
          data-v-2964abc9=""
          style="background-color: rgba(255, 255, 255, 0.9); display: none;"
        >
          <div
            class="velmld-spinner"
            data-v-2964abc9=""
          >
            
            <svg
              data-v-2964abc9=""
              data-v-2964abc9-s=""
              enable-background="new 0 0 40 40"
              height="50px"
              id="loader-1"
              version="1.1"
              viewBox="0 0 40 40"
              width="50px"
              x="0px"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
              y="0px"
            >
              <path
                d="M20.201,5.169c-8.254,0-14.946,6.692-14.946,14.946c0,8.255,6.692,14.946,14.946,14.946
      s14.946-6.691,14.946-14.946C35.146,11.861,28.455,5.169,20.201,5.169z M20.201,31.749c-6.425,0-11.634-5.208-11.634-11.634
      c0-6.425,5.209-11.634,11.634-11.634c6.425,0,11.633,5.209,11.633,11.634C31.834,26.541,26.626,31.749,20.201,31.749z"
                fill="#000"
                opacity="0.2"
              />
              <path
                d="M26.013,10.047l1.654-2.866c-2.198-1.272-4.743-2.012-7.466-2.012h0v3.312h0
      C22.32,8.481,24.301,9.057,26.013,10.047z"
                fill="#000"
              >
                <animatetransform
                  attributeName="transform"
                  attributeType="xml"
                  dur="0.6s"
                  from="0 20 20"
                  repeatCount="indefinite"
                  to="360 20 20"
                  type="rotate"
                />
              </path>
            </svg>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        category="[object Object]"
        class="m-2"
        style=""
      >
        <div>
          <div
            class="card text-white"
          >
            <img
              class="card-img category-card"
              src="http://storage/image/path/xxxx.jpg"
              style="max-width: 20rem; height: 200px; filter: brightness(0.65);"
            />
            <div
              class="card-img-overlay"
            >
              <h4
                class="card-title"
              >
                公務員
              </h4>
              <p
                class="card-text"
              >
                <span>
                  役所、消防署、官僚、国税専門官
                </span>
                <a
                  class="stretched-link"
                  href="#"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        category="[object Object]"
        class="m-2"
        style=""
      >
        <div>
          <div
            class="card text-white"
          >
            <img
              class="card-img category-card"
              src="http://storage/image/path/xxxx.jpg"
              style="max-width: 20rem; height: 200px; filter: brightness(0.65);"
            />
            <div
              class="card-img-overlay"
            >
              <h4
                class="card-title"
              >
                営業
              </h4>
              <p
                class="card-text"
              >
                <span>
                  コンサルティングフォーム、専門事務所
                </span>
                <a
                  class="stretched-link"
                  href="#"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
